mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -1936,6 +1937,100 @@ def instructions(self) -> str:
         return CONVERSATIONAL_TOOL_CALL_EFFICIENCY_PROMPT
 
 
+@experimental(version="3.8.0")
+@format_docstring(_MODEL_API_DOC)
+class ConversationalRoleAdherence(BuiltInSessionLevelScorer):
+    """
+    Conversational role adherence evaluates whether an AI assistant maintains its assigned
+    role throughout a conversation.
+
+    This scorer analyzes the assistant's responses for role adherence concerns including:
```

Comment:
nit: same comment as before about keeping this generic if we ever change the implementation later